### PR TITLE
Add more verbose info in fe log and profile

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：
This PR is to add som key metrics into profile. Those metrics are critical when running query on external table: hive/iceberg/hudi. 

image

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 8 / 11 (72.73%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|5|7|71.43%|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|3|4|75.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -278,6 +284,8 @@ public void execute() throws Exception {
         UUID uuid = context.getQueryId();
         context.setExecutionId(UUIDUtil.toTUniqueId(uuid));
         SessionVariable sessionVariableBackup = context.getSessionVariable();
+        LOG.info("begin execute query id = " + DebugUtil.printId(uuid));
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:287]: It's better to add to the LOG.debug()

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:287]: OK. but after second though,  I see there are begin/end time on profile, so there is no need to log it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 32 / 63 (50.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|7|42.86%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|25|50|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|4|6|66.67%|

**Review by Youngwb — APPROVED** [review]

**Comment by Seaven** [comment]
I don't think it's a pretty design, external table logic and related metrics should not be exposed to executors, your code looks just for debug, I think you should think systematically about this design and you can discuss with @silverbullet233

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java [hunk 2]
```diff
@@ -133,9 +134,13 @@
 
     protected DumpInfo dumpInfo;
 
+    protected HiveMetaCache.ProfileStats hiveMetaCacheProfileStats;
+
     // The related db ids for current sql
     protected Set<Long> currentSqlDbIds = Sets.newHashSet();
 
+    protected int plannerRunningTime = 0;
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java:142]: maybe `long` is better?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java [hunk 3]
```diff
@@ -28,6 +30,36 @@
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 public class HiveMetaCache {
+    public static class ProfileStats {
+        int onPartitionKeys = 0;
+        int onPartitions = 0;
+        int onPartitionStats = 0;
+        int onTableStats = 0;
+        int onColumnStats = 0;
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:38]: Package level access?

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java (hunk 2 above)
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java:142]: better to take the unit in the name. Such as planRunningMs planRunningSeconds

**Pull request closed without merging**